### PR TITLE
Fix for handling back press on Authentication Activity (#292)

### LINKED ISSUE #292: Getting Introduction Activity on pressing back button
Introduction Activity is not being shown on the first launch when the user is not logged in. It gets added to Back Stack and is replaced by Authentication Activity.

Can anyone confirm that this is a bug or done intentionally?

### PR DESCRIPTION
**Pull request by adi23arora**
This fixes #292

### CODE REVIEW

File: src/main/java/org/amahi/anywhere/activity/AuthenticationActivity.java [hunk 1]
```diff
@@ -308,4 +308,9 @@ protected void onPause() {
 
         BusProvider.getBus().unregister(this);
     }
+
+    @Override
+    public void onBackPressed() {
+        finishAffinity();
```
Comment (csoni111) [inline src/main/java/org/amahi/anywhere/activity/AuthenticationActivity.java]: I believe this is unnecessary.

File: src/main/java/org/amahi/anywhere/activity/IntroductionActivity.java [hunk 2]
```diff
@@ -59,5 +59,6 @@ public void onSkipPressed(Fragment currentFragment) {
 
     private void launchTv() {
         startActivity(new Intent(this, NavigationActivity.class));
```
[the remark this answers is not in the document]
Comment (adi23arora) [reply src/main/java/org/amahi/anywhere/activity/IntroductionActivity.java]: I also wondered why it was done this way?

File: src/main/java/org/amahi/anywhere/activity/AuthenticationActivity.java (hunk 1 above)
Comment (adi23arora) [reply src/main/java/org/amahi/anywhere/activity/AuthenticationActivity.java]: There are 2 cases : 
1. When the user is not logged in and its also the first time launch - In this case, we will have Introduction Activity and Navigation Activity in the back stack. On pressing back button first Introduction Activity will be popped then on second press Navigation Activity (with a blank screen).
2. When the user is not logged in and not a first time launch - In this, we would have Navigation Activity in back stack.

So this finishes all activities in stack and works well both of these cases

**Comment by adi23arora** [comment]
@csoni111 Combination of these 2 commits worked well to solve all the problems arising. If there's any other way or approach to this, I would love to discuss and implement.

File: src/main/java/org/amahi/anywhere/activity/AuthenticationActivity.java (hunk 1 above)
Comment (csoni111) [reply src/main/java/org/amahi/anywhere/activity/AuthenticationActivity.java]: Ok I understand the use cases. This seems fine.

**Pull request merged**